lib/assessment.js

```diff
@@ -39,9 +45,19 @@ var newArray2 = function newArray2(length, defaultsFunction) {
 // The function you write should return
 // the object passed in as an argument or the object created
 var addMethod = function addMethod(propertyName, method, obj) {
-
+  var Method = {
+    propertyName: propertyName,
+    method: method,
+    obj: obj
+  };
```

Comment:
This builds a new object that has a reference to both `obj` and `method`. It doesn't satisfy the question, but it does do something important right: it places a method on an object.

Here's how I would have solved this:

```js
var addMethod = function addMethod(propertyName, method, obj) {
  if(!obj) {
    obj = {};
  }

  obj[propertyName] = method;

  return obj;
}
```

1/2